arch/cortex-m4/src/lib.rs

```diff
@@ -351,11 +352,8 @@ pub unsafe extern "C" fn hard_fault_handler() {
         // hard fault occurred in an app, not the kernel. The app should be
         //  marked as in an error state and handled by the kernel
         asm!(
-            "ldr r0, =SYSCALL_FIRED
-              mov r1, #1
-              str r1, [r0, #0]
-
-              ldr r0, =APP_FAULT
+            "ldr r0, =APP_FAULT
+              mov r1, #1 /* Fault */
```

Comment:
Where are we actually writing to the `SYSCALL_FIRED` variable?

Reply:
I'm not sure what you mean.

Reply:
Ah, same place we were before: https://github.com/tock/tock/pull/1113/files/a6c2fe1c10227dba72dbe63499f90f8d800d559a#diff-3c4f4f49e6d8b6e9ec96c45e8c737637R146

I think marking that a syscall happened in the fault handler was a hack before.

Reply:
Okay, that makes a lot of sense now. It was just a hack for routing the failure presumably.